### ISSUE TITLE
Paper key generation is confusing

### ISSUE THREAD

**Issue by rcsheets** [keybase/keybase-issues#2654]
Using the Linux GUI, in the paper key generation process, it's not clear:
- Why this is happening
- What the paper key will be used for in the future
- How sensitive the paper key is

Is it equivalent to my keybase account passphrase? Is it equivalent to my GPG passphrase? Is it equivalent to any of my GPG key material? All of this is very unclear. All that's stated is that it's important and I need to write it down.

**Comment by rcsheets** [comment]
I found [some information on paper keys](https://keybase.io/docs/kbfs#paper-keys) in the kbfs docs that says it's a "full-powered private key". I'm beginning to think this operation should not be required when setting up the keybase app. At the very least, it should be made much clearer what's going on.

**Comment by juanminm** [comment]
> What the paper key will be used for in the future
Should you lose all your devices or have them compromised, you can use a paperkey to provision new devices. So...

> I'm beginning to think this operation should not be required when setting up the keybase app

unless you want to risk the possibility of having to start from scratch in a doomsday scenario, you may want to have one or more safeguarded paperkeys at least until the iOS and Android apps are released.

**Comment by rcsheets** [comment]
> Should you lose all your devices or have them compromised, you can use a paperkey to provision new devices.

This should be explained by the UI. Otherwise the user has no idea what they're doing, like I didn't.

> unless you want to risk the possibility of having to start from scratch in a doomsday scenario

Start what from scratch? Lose access to my keybase username? I still don't understand the consequences of losing (or losing control over) my provisioned device and also losing my paper key. What's being safeguarded?

**Comment by maxtaco** [comment]
Encrypted kbfs data is only recoverable by one of your secret keys.

**Comment by rcsheets** [comment]
> Encrypted kbfs data is only recoverable by one of your secret keys.

Does that include my GPG key?

**Comment by maxtaco** [comment]
No currently kbfs data is not encrypted for gpg keys

**Comment by stevesbrain** [comment]
@maxtaco So does that mean that the paper key is basically used as a "master" key for all KBFS data, and any "subkey" (i.e. device it provisions) is able to also decrypt the same data? I'm presuming this paperkey can therefore also sign files as me?

This seems to be an option that shouldn't be forced upon users if so. For example, I have a YubiKey which stores my only accessible copy of my private signing and encryption keys - everything else lives offline in an encrypted volume, and never goes online (I have an offline machine that performs all my master key modifications). In this case, if I am forced to generate a paper key, and I lose control of that, then an adversary will be able to sign things as me, as well as decrypt any of my KBFS data etc (without a password required). If I was able to use only the keys that I'm used to using, then I can guarantee I always retain full control of these.

Is there a reason we would not (could not?) just use our existing (sub)keys and trust that the user appropriately backs these up (or at least give that option)? Am I correct in understanding that the paperkey's private component lives on keybase + is distributed by keybase, and the key we have recorded is essentially the passphrase used to unlock it?

**Comment by maxtaco** [comment]
You are mostly correct. The paper key is the private key. Our servers never
see any version of this secret key, encrypted or otherwise. Using modern
crypto (unlike standard PGP), only ~100 bits of entropy are needed to give
full secret keys. If this is not to your liking then simply revoke the
paper key.  But then you risk data loss if you lose all of your other keys.

**Comment by stevesbrain** [comment]
@maxtaco Thanks for the reply - could you confirm if the paper key is only the private key for the chat service + FS (as I suspect it is), or if this is a key capable of decrypting anything destined for the overall PGP identity? I don't see how the latter is possibly, given my understanding of PGP, but I just want to be sure. Thanks!

**Comment by maxtaco** [comment]
Yes, it's a private key for signing and decryption of KBFS and chat data.

**Comment by stevesbrain** [comment]
Thanks for confirming!
